packages/react-app/components/notifications/OnboardingBatch.jsx

```diff
@@ -1,38 +1,133 @@
-import { Box, CloseButton, Flex, Text, Button, Link, Image, Spacer, Tooltip } from "@chakra-ui/react";
 import React from "react";
+import {
+  Box,
+  CloseButton,
+  Flex,
+  Text,
+  Button,
+  Link,
+  Image,
+  Spacer,
+  Tooltip,
+  useToast,
+  useColorModeValue,
+} from "@chakra-ui/react";
+import { getUpdateBatchSignMessage, postUpdateBatch } from "../../data/api";
+import { useUserAddress } from "eth-hooks";
 
 const NEXT_BATCH_TELEGRAM = process.env.NEXT_PUBLIC_BATCH_TELEGRAM;
 const NEXT_BATCH_DATE = process.env.NEXT_PUBLIC_BATCH_DATE;
+const NEXT_BATCH = process.env.NEXT_PUBLIC_BATCH_NUMBER;
+
+const OnboardingBatch = ({ notification, onMarkAsRead, builder, userProvider, onUpdate }) => {
+  const address = useUserAddress(userProvider);
+
+  const hasTelegram = builder?.socialLinks?.telegram;
+
+  const toast = useToast({ position: "top", isClosable: true });
+  const toastVariant = useColorModeValue("subtle", "solid");
+
+  const handleUpdateBatch = async () => {
```

Comment:
This is a very weird UX (blank tab, auto redirect...). Let's iterate on it a bit.

I think a better approach is:

1. Check if the user is already in a batch.
 (-> YES. Show "Join telegram" button and link to the TG channel (as it currently works in the live site))
2. NO. Start the "Join Batch" workflow
- We have a (yellow?) button that says "Request Telegram access" or similar.
- Sign a message (you are doing this already)
- User gets tagged as candidate  in the active batch
- It shows a success message
- The banner updates with "Join telegram" button.

IF the user refreshes, or come back later.... since they are already in the batch it'll show the "Join Telegram"

Reply:
The reason I chose that unconventional UX is because I wanted everything in one seamless workflow, without splitting it into two separate steps. This way, users wouldn’t mistakenly think they had already joined the Telegram channel just by signing the message.

I agree that your workflow is def. cleaner and the blank tab is not the nicest.